services/api/src/resources/deployment/resolvers.js

```diff
@@ -166,6 +166,26 @@ const getDeploymentByRemoteId = async (
   return injectBuildLog(deployment);
 };
 
+const getDeploymentUrl = async (
+  { id, environment },
+  args,
+  { sqlClient, hasPermission },
+) => {
+
+
+  const domain = process.env.LAGOON_UI || 'localhost';
```

Comment:
we have a system in the api already that allows Lagoon to autodiscover the URLs of other services, see
https://github.com/amazeeio/lagoon/blob/1386194da63bd267c47a3534a3d375fe474caa3b/services/api/src/clients/keycloakClient.js#L21-L26

Can you implement this here?